sw/host/cryptotest/testvectors/parsers/wycheproof_ecdh_parser.py

```diff
@@ -0,0 +1,128 @@
+        for test in group["tests"]:
+            logging.debug(f"Parsing tcId {test['tcId']}")
+            # Reject tests with invalid ASN
+            if "InvalidAsn" in test["flags"]:
+                logging.info(
+                    f"Skipped tcId {test['tcId']}: Tagged as Invalid ASN."
+                )
+                continue
+            # Wycheproof flags one (valid) test using a compressed key
+            # encoding as "acceptable".  Because all other
+            # "acceptable" tests should be rejected, we throw this one
+            # out.
+            if "CompressedPoint" in test["flags"] and test["result"] == "acceptable":
+                logging.info(
+                    f"Skipped tcId {test['tcId']}: Valid compressed key test marked 'acceptable'"
+                )
+                continue
+
+            test_vec = {
+                "vendor": "wycheproof",
+                "test_case_id": test['tcId'],
+                "algorithm": "ecdh",
+                "curve": EC_NAME_MAPPING[group["curve"]],
+                "d": list(bytes.fromhex(test["private"])),
+                "z": list(bytes.fromhex(test["shared"])),
+            }
+
+            # Parse ASN encoded public key
+            if group["encoding"] == "asn":
+                try:
+                    public_key = ECC.import_key(bytes.fromhex(test["public"]))
+                except ValueError as e:
+                    logging.info(
+                        f"Skipped tcId {test['tcId']}: ValueError when parsing DER sequence: {e}."
+                    )
+                    continue
+                if public_key.curve != EC_NAME_MAPPING[group["curve"]]:
+                    logging.info(
+                        f"Skipped tcId {test['tcId']}: Curve in DER did not match test vector."
+                    )
+                    continue
+                test_vec["qx"] = list(public_key.pointQ.x.to_bytes())
+                test_vec["qy"] = list(public_key.pointQ.y.to_bytes())
+            else:
+                logging.info(
+                    f"Skipped tcId {test['tcId']}: unsupported encoding '{group['encoding']}'."
+                )
+                continue
+
+            # Parse the expected result
+            if test["result"] == "valid":
+                test_vec["result"] = True
+            elif test["result"] == "invalid":
+                test_vec["result"] = False
+            elif test["result"] == "acceptable":
+                # Err on the side of caution and reject "acceptable" key derivations
+                test_vec["result"] = False
+            else:
+                raise RuntimeError(f"Unexpected result type {test['result']}")
+
+            test_vectors.append(test_vec)
+
+    return test_vectors
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--src',
+        metavar='FILE',
+        type=argparse.FileType('r'),
+        help='Read test vectors from this JSON file.'
```

Comment:
Nit: keep the tone of the help messages consistent with the rest of the args and with the python scripts in general in the repo.
```suggestion
        help="Test vector input file to parse"
```